### PR TITLE
Lock color

### PR DESCRIPTION
**Pull request by bradybridges**
#### What's this PR do?
This PR add basic palette generation functionality. Allows the user to generate a new palette, lock colors, and save palettes to a specific project. There currently is no indication that a palette saves, so will need to go back and clear the form and display a message of some type. Palette is added to db and store at this time, so it persists.
#### Where should the reviewer start?
Make sure generate palette is working, locking colors is working, and that on load the store is updated with currrent projects and palettes.
#### Any background context you want to provide?
No styling added currently so looks pretty rough, but functions. Currently only using GET from projects/palettes, and POST palettes. Other apiCalls have not been tested. Will need to update save palette form to use state for input and select value.

### CODE REVIEW

**Comment by bradybridges** [comment]
Screen Shot 2019-12-07 at 8 24 03 AM

**Comment by bradybridges** [comment]
Screen Shot 2019-12-07 at 8 24 03 AM

**Comment by bradybridges** [comment]
Screen Shot 2019-12-07 at 8 24 49 AM

File: src/apiCalls.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  const response = await fetch('https://color-picker-backend.herokuapp.com/api/v1/palettes');
+  if(!response.ok) {
+    return Error('Failed to fetch palettes');
+  }
+  const palettes = await response.json();
+  return palettes;
+};
+
+export const getProject = async (name) => {
+  const response = await fetch(`https://color-picker-backend.herokuapp.com/api/v1/projects/${name}`);
```
Comment (slkarsh) [inline src/apiCalls.js:20]: Will update this for the reformatted endpoint to include id instead of name

**Pull request merged**